src/Symfony/Component/Security/Http/Firewall/ContextListener.php

```diff
@@ -292,7 +292,7 @@ protected function refreshUser(TokenInterface $token)
             return null;
         }
 
-        throw new \RuntimeException(sprintf('There is no user provider for user "%s".', $userClass));
+        throw new \RuntimeException(sprintf('There is no user provider for user "%s".%s', \get_class($user), count($this->providers) > 0 ? ' Shouldn\'t the "supportsClass()" method of your user provider returns true for this classname?' : ''));
```

Comment:
`returns` => `return`?

Reply:
Isn't "the .. method of your user provider" third person? I agree that "returns" sounds a bit weird, but I feel like gramatically it's more correct. (but I'm not close to native english)

Reply:
I think the verb-subject agreement applies to `should` here (can be replaced by `does it return?`.)
More importantly: shouldn't `$this->providers` be `$this->userProviders` here? 
Also `iterable` is not always count-able. Let's skip the check, you won't reach this statement if `$this->providers <= 0` anyway, same for other patched classes.
>

Reply:
Oh, alright. Updated all 3 cases